### PR TITLE
Add `padding_mode` for RandomElasticTransform augmentation

### LINKED ISSUE #1192: Exposing `padding_mode` in elastic transforms.
## 🚀 Feature

Exposing `padding_mode` in `RandomElasticTransform` and `elastic_transform2d`.
## Motivation

 In `F.grid_sample` as used in kornia's elastic transforms, the default setting `padding_mode = 'zeros'` leaves unnatural black artifacts in the newly created borders of transformed images. 
## Pitch

Exposure of `padding_mode` in elastic transform functions/classes (as it seems has been done for affine warping functions/classes https://github.com/kornia/kornia/pull/640) would allow for selection of more natural settings like 'reflection'.

Thanks and outstanding work!

### PR DESCRIPTION
**Pull request by dobosevych**
#### Changes
Adds padding_mode  for RandomElasticTransform augmentation

Fixes #1192 

#### Type of change
- [x] 📚  Documentation Update
- [x] 🔬 New feature (non-breaking change which adds functionality)
- [x] 📝 This change requires a documentation update

#### Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] Did you update CHANGELOG in case of a major change?

### CODE REVIEW

**Review by edgarriba — APPROVED** [review]

File: kornia/geometry/transform/elastic_transform.py [hunk 1]
```diff
@@ -17,6 +18,7 @@ def elastic_transform2d(
     alpha: Tuple[float, float] = (1.0, 1.0),
     align_corners: bool = False,
     mode: str = 'bilinear',
+    padding_mode: Union[str, int, SamplePadding] = SamplePadding.ZEROS.name
```
Comment (edgarriba) [inline kornia/geometry/transform/elastic_transform.py:21]:
```suggestion
    padding_mode: str = 'zeros'
```
torchscript doesn't support well yet `Union` type

**Review by edgarriba — CHANGES_REQUESTED** [review]

File: kornia/geometry/transform/elastic_transform.py (hunk 1 above)
Comment (odobosevych) [reply kornia/geometry/transform/elastic_transform.py:21]: just change to str for now?

Comment (edgarriba) [reply kornia/geometry/transform/elastic_transform.py:21]: Yeah

File: kornia/augmentation/augmentation.py [hunk 2]
```diff
@@ -2168,6 +2170,7 @@ def __init__(
         alpha: Tuple[float, float] = (1.0, 1.0),
         align_corners: bool = False,
         mode: str = 'bilinear',
+        padding_mode: Union[str, int, SamplePadding] = SamplePadding.ZEROS.name,
```
Comment (edgarriba) [inline kornia/augmentation/augmentation.py:2173]: Types should go according to th fucntional, this should be string

File: kornia/geometry/transform/elastic_transform.py [hunk 3]
```diff
@@ -17,6 +18,7 @@ def elastic_transform2d(
     alpha: Tuple[float, float] = (1.0, 1.0),
     align_corners: bool = False,
     mode: str = 'bilinear',
+    padding_mode: str = SamplePadding.ZEROS.name
```
Comment (edgarriba) [inline kornia/geometry/transform/elastic_transform.py:21]:
```suggestion
    padding_mode: str = 'zeros',
```

File: kornia/augmentation/augmentation.py (hunk 2 above)
Comment (odobosevych) [reply kornia/augmentation/augmentation.py]: fixed

File: kornia/geometry/transform/elastic_transform.py (hunk 3 above)
Comment (dobosevych) [reply kornia/geometry/transform/elastic_transform.py]: fixed, I will run test locally to check that everything passes to don't disturb you again

**Comment by dobosevych** [comment]
I run tests locally and this one should work. Only some jit test/enhance/test_histogram.py fails, but I think it should be, as that

**Comment by dobosevych** [comment]
I don't like that we have constants.py, but aren't using it in most cases. I would propose to create something like a static factory that will allow converting between different libs formats in constatns.py and use it.

**Review by edgarriba — APPROVED** [review]
LGTM

**Comment by edgarriba** [comment]
@dobosevych in case you are interested in elastic transform - there's this pending PR to improve the functionality https://github.com/kornia/kornia/pull/1050

**Comment by dobosevych** [comment]
Ok. I was also thinking of taking a look into this https://github.com/kornia/kornia/issues/1143, as a first step planned to create a list of ImageCorruption that aren't been implemented yet.

**Comment by edgarriba** [comment]
yes, definitely - check also this one https://github.com/kornia/kornia/issues/941. There's a the Plasma group that we wanted to add for a while with @anguelos . Feel free to choose any of them,

**Pull request merged**